docs/design/solution-architecture.md

```diff
@@ -0,0 +1,14 @@
+# High Level Solution Architecture
+
+The following figure illustrates briefly how the overall solution works. ![Apex Solution](./apex-architecture.png).
```

Comment:
Is there a source for the diagram? We'll need something that others can modify and update if needed.

Something to check out is https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/ which would have the diagram automatically generated. That would be ideal if you can get it to display what you want.

Reply:
I agree editable diagrams are preferable to images where possible. That mermaid utility looks interesting, thanks for the pointer. However it seems more suited for simpler flowcharts than arbitrary diagrams like what we have in the PR currently with clouds and arbitrary end to end curved lines like the wireguard tunnel.

Reply:
I looked around a bit. I see some example that people have made that look pretty good. Here's one that's networking oriented.

https://github.com/mermaid-js/mermaid/issues/1227#issuecomment-1192060275